src/main.py

```diff
@@ -278,6 +287,145 @@ def grant(self, sess, comment, grantee, badge):
             investor.badges = json.dumps(badge_list)
             return comment.reply_wrap(message.modify_grant_success(grantee, badge))
 
+    @req_user
+    def firm(self, sess, comment, investor):
+        if investor.firm == 0:
+            return comment.reply_wrap(message.firm_none_org)
+
+        firm = sess.query(Firm).\
+            filter(Firm.id == investor.firm).\
+            first()
+
+        ceo = "/u/" + sess.query(Investor).\
+            filter(Investor.firm == firm.id).\
+            filter(Investor.firm_role == "ceo").\
+            first().\
+            name
+        execs = concat_names(
+            sess.query(Investor).\
+                filter(Investor.firm == firm.id).\
+                filter(Investor.firm_role == "exec").\
+                all())
+        traders = concat_names(
+            sess.query(Investor).\
+                filter(Investor.firm == firm.id).\
```

Comment:
Don't we need to join the tables here?

Reply:
We already did a query to retrieve the firm data above, so it seems like doing the user queries without the join and just using querying by firm id would be a little faster. Or is there a better way of doing it that I'm missing?